docs/pbc/all/payment-service-provider/202410.0/base-shop/third-party-integrations/payone/app-composition-platform-integration/payone-acp-app.md

```diff
@@ -87,11 +87,169 @@ When customers pay with PayPal, a shop owner can do the following:
+* clientId
+* merchantId
+7. When the customer clicks "Complete Purchase" the modal closes and the customer data is requested.
+8. After this the customer should be redirected to the summary page.
+9. Here he should see his address data that was received from PayPal, and he can change the data or submit it which makes an API request to persist the order in the Back Office: `glue.mysprykershop.com/checkout`.
+10. The customer is redirected to the application's success page.
+11. PayOne app confirms the pre-order payment using the plugin: `\Spryker\Zed\Payment\Communication\Plugin\Checkout\PaymentConfirmPreOrderPaymentCheckoutPostSavePlugin`.
+    The `order_reference` is passed to the PayOne app to be connected with `transaction_id`.
+12. PayOne app processes the payment and sends a `PaymentUpdated` message to Spryker.
+13. Depending on payment status, one of the following messages is returned through an asynchronous API:
+*  Payment is successful: `PaymentAuthorized` message.
+*  Payment is failed: `PaymentAuthorizationFailed` message.
+14. Further on, the order is processed through the OMS.
+
+All payment related messages mentioned above are handled by `\Spryker\Zed\Payment\Communication\Plugin\MessageBroker\PaymentOperationsMessageHandlerPlugin`, which is registered in `MessageBrokerDependencyProvider`.
+
+
+### Example of the headless checkout with PayOne
+
+The PaymentMethods which are returned by the `/checkout-data?include=payment-methods` endpoint are used to render the express-checkout buttons e.g. on the Product Detail Page or the Cart Page. The customer can select the payment method and proceed to the payment page.
+
+You can read more details about this in the [Payment Method Strategies](https://documentation.spryker.com/docs/pbc/all/payment-service-provider/202410.0/base-shop/payment-method-strategies.html) document.
+
+When the custome clicks on "Pay with PayPal Express" button, the `InitializePreOrderPayment` Glue API endpoint is called to initialize the payment.
+
+#### Pre-order payment initialization
+
+```JS
+
+async initializePreOrderPayment() {
+    const requestData = {
+      data: {
+        type: 'payments',
+        attributes: {
+          quote: QUOTE_DATA,
+          payment: {
+            amount: GRAND_TOTAL, // You will get it through the `/checkout-data?include=carts` endpoint
+            paymentMethodName: 'PayPal Express', // taken from /checkout-data?include=payment-methods
+            paymentProviderName: 'PayOne',  // taken from /checkout-data?include=payment-methods
+          },
+          preOrderPaymentData: {
+            "transactionId": this.transactionId, // This is empty in the first request but has to be used in further requests
+          },
+        },
+      },
+    };
+
+    const responseData = await this.fetchHandler(`GLUE_APPLICATION_URL/payments`, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: `Bearer ACCESS_TOKEN`,
```

Comment:
```suggestion
        Authorization: `Bearer ACCESS_TOKEN`, // taken from /access-tokens
```

also example for guest cart should be provided with `X-Anonymous-Customer-Unique-Id`

Reply:
It's described as for Stripe the line below mentions it. I need to update the SPA to also work in GUest mode, then I will add examples to both PayOne and Stripe.